src/editor/editor.cpp

```diff
@@ -27,15 +29,22 @@ WorldPosition tilemap_position = {0.0, 0.0};
 
 int tilemap_row_mouse_on_display = 0;
 int tilemap_col_mouse_on_display = 0;
-int window_mouse_x = 0;
-int window_mouse_y = 0;
+int imgui_window_mouse_x = 0;
+int imgui_window_mouse_y = 0;
 int tileset_width = 1;
 int tileset_height = 1;
 int tileset_channels = 0;
 float logical_mouse_x = 0;
 float logical_mouse_y = 0;
 float relative_tilemap_mouse_x;
 float relative_tilemap_mouse_y;
+v2d panning_offset = {0, 0};
+v2d imgui_tilemap_position = {tilemap_position.x, tilemap_position.y};
+v2d imgui_axis_position = {0, 0};
+v2d imgui_offset_tilemap = {0, 0};
+v2d imgui_offset_axis = {0, 0};
+v2d global_offset = {0, 0};
```

Comment:
what is this the global offset of?

Reply:
The entirety of what's being rendered to the im_gui window. See the explanation in the image (Hopefully its clear enough just message if it isnt).
![drawing_lmao](https://user-images.githubusercontent.com/31080326/194726862-f2f42806-61fc-4142-be24-2488abddf93e.png)

Reply:
renamed global_offset to imgui_window_global_offset for further clarification